backend/production-optimiser-api/src/main/java/it/polimi/productionoptimiserapi/controllers/OptimizationModelController.java

```diff
@@ -16,20 +21,30 @@
 
 @RestController
 @RequestMapping("/api/models")
+@RequiredArgsConstructor
 public class OptimizationModelController {
 
   private final OptimizationModelService optimizationModelService;
-
-  public OptimizationModelController(OptimizationModelService optimizationModelService) {
-    this.optimizationModelService = optimizationModelService;
-  }
+  private final UserService userService;
 
   @PostMapping
   @PreAuthorize("hasRole('ADMIN')")
   public ResponseEntity<OptimizationModel> create(
       @Valid @RequestBody OptimizationModelDTO optimizationModelDTO) {
     OptimizationModel om =
         this.optimizationModelService.saveOptimizationModel(optimizationModelDTO);
+
+    // Add the new optimization model to all admins
+    userService.getUsers().stream()
+        .filter(user -> user.getRole().equals(UserRole.ADMIN))
+        .forEach(
+            user -> {
+              Set<String> updatedModelIds = new HashSet<>(user.getOptimizationModelIds());
+              updatedModelIds.add(om.getId());
+
+              userService.updateUser(user.getId(), null, null, null, updatedModelIds);
+            });
```

Comment:
this creates a lot of problems due to a way code is structured right now, if i include UserService inside OptimizationModelService then we have circular dependency, im not happy with current solution aswell but it would cause whole refactoring of both user and model services 

we should maybe create jira ticket for this refactoring in this spring

Reply:
thats true, i meant maybe adding a method to user service which does the same thing as this part of the code, just so that its in the correct layer

we can add that

Reply:
oh yeah didnt think of that.. fixed